### PR TITLE
[Arm64/Win] Revise JIT_MemSet

### PR DESCRIPTION
**Pull request by sdmaclea**
@dotnet/arm64-contrib @jkotas This is the Windows equivalent of #11217 which merged

### CODE REVIEW

**Comment by BruceForstall** [comment]
@janvorli

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT Arm64 Checked

**Comment by BruceForstall** [comment]
@dotnet/arm64-contrib @jkotas @janvorli PTAL

**Comment by BruceForstall** [comment]
@dotnet-bot test this please

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT Arm64 Checked

**Comment by janvorli** [comment]
@sdmaclea the Windows ARM64 build failed in the CI while building the crthelpers.asm with an unknown error somewhere.

**Comment by sdmaclea** [comment]
@janvorli I'll take a look.  My guess is asm doesn't know about the zva registers.

File: src/vm/arm64/crthelpers.asm [hunk 1]
```diff
@@ -87,68 +134,89 @@
...
-    mov         x10,x0
-    add         x8,x10,x11 lsl #3
+    ands        w8, w1, #0xff
+    mrs         x3, DCZID_EL0                      ; x3 = DCZID_EL0
+    mov         x6, #4
+    lsr         x11, x2, #3                        ; x11 = count >> 3
+
+    orr         w8, w8, w8, lsl #8
+    and         x5, x3, #0xf                       ; x5 = dczid_el0.bs
+    csel        x11, x11, xzr, eq                  ; x11 = (val == 0) ? count >> 3 : 0
```
Comment (sdmaclea) [inline src/vm/arm64/crthelpers.asm]: @rahku The windows arm64 assembler does not like this instruction.  Do you know how to write it?

Comment (BruceForstall) [inline src/vm/arm64/crthelpers.asm]: fyi, @rahku is no longer working on the CLR, so may not monitor GitHub. Maybe @janvorli has an idea.

Comment (janvorli) [inline src/vm/arm64/crthelpers.asm]: @sdmaclea - it is not clear which of the three instructions above you are referring to. I assume it is the csel, right?

Comment (sdmaclea) [inline src/vm/arm64/crthelpers.asm]: Yes the assembler is complaining about the `csel`
```
C:\DATA\clr\try_new_toolchain\coreclr\bin\obj\Windows_NT.arm64.Checked\src\vm\wks\CrtHelpers.asm(3498) : error A2506: Unknown error
    csel        x11, x11, xzr, eq                  ; x11 = (val == 0) ? count >> 3 : 0
```

Should I inject the raw encoding? How?

Comment (janvorli) [inline src/vm/arm64/crthelpers.asm]: I am asking around to see if I can find out the syntax that the windows assembler expects.

Comment (sdmaclea) [inline src/vm/arm64/crthelpers.asm]: Thanks.  Is the windows arm64 assembler doc available publicly?

Comment (janvorli) [inline src/vm/arm64/crthelpers.asm]: @sdmaclea I got an answer from the ARM64 assembler team. The syntax should be
`cseleq x11, x11, xzr`

Comment (sdmaclea) [inline src/vm/arm64/crthelpers.asm]: @janvorli Thanks.  That does not match the current ARMv8 ARM, but if it compiles it is good enough.  I'll push an update.

**Comment by sdmaclea** [comment]
@janvorli Thanks, changing the csel form fixed the assembler error.  This is passing all tests and can be reviewed/merged.

**Pull request merged**